### DISCUSSION TITLE
Uhhh i wanted to host a discord bot

### DISCUSSION THREAD

**Discussion by kakarot-dev** [vercel/vercel#6137] [General]
I wanted to ask but can we host discord bots here?

**Comment by kakarot-dev** [comment]
Still waiting for a reply …

**Comment by jayhale** [comment]
If your bot uses just the [HTTP API](https://discord.com/developers/docs/reference#http-api), it would likely be easily implementable. It would be helpful to have some sort of persistence layer so that you don't have to authenticate every time you want to make a new call.

There is a [python implementation](https://discordpy.readthedocs.io/en/stable/api.html) of the API that would run fine on Vercel.

If you want to use the Websockets API or have voice support, Vercel might not be the easiest solution - you might try something like DigitalOcean.
